data-raw/process_data_sets.R

```diff
@@ -272,3 +272,33 @@ pennies_resamples <-
   select(replicate, everything()) %>%
   unnest(cols = c(data))
 usethis::use_data(pennies_resamples, overwrite = TRUE)
+
+#-------------------------------------------------------------------------------
+# Babies: Documented in R/babies.R
+#-------------------------------------------------------------------------------
+# Population of 1236 babies from
+# https://wjhopper.github.io/SDS-201/data/babies.csv
+babies <- "https://wjhopper.github.io/SDS-201/data/babies.csv" %>%
+  read_csv() 
+
+# Creating new date variable
+babies <- babies %>%
+  mutate(birthday = date)
+
+# Creating new dates
+create <- tibble(
+  days = format(seq(as.Date("1961-09-12"), as.Date("1962-09-11"), by="days"), format="%m-%d-%Y"),
+  date_days = c(1350:1714)
+) %>% 
+  mutate(merged = paste(days, date_days))
+
+# Updating birthday in babies dataset
+val = 0
+index = 1
+for (i in babies$birthday) {
```

Comment:
What is happening here?

Reply:
Birth date in this dataset is noted by a 4 digit code where 1096 is January 1st, 1961. We thought it would be helpful if the user could see the actual birthdate without have to calculate days from January 1st, 1961. This loop creates the new birthday variable.

Reply:
(In the best way I know how.. I'm sure there's more efficient ways)

Reply:
The loop goes through each row of the `babies` dataset, updates the `birthday` column to a dd/mm/yyyy date. The loop uses the `create` tibble that has a column called `merged` with the 4 digit date and its corresponding dd/mm/yyyy date, combined into one (i.e. 09-12-1961 1350). The loop starts with the first row of `babies$date` (i.e. 1411) identifies which row of `merged` has the 4 digit date at the end of the string (i.e. 11-12-1961 1411), and then it replaces the value for `birthday` with the first part of the string (i.e. 11-12-1961). Then it moves to the next row in `babies$date` .

Reply:
OK. So note that 365 * 3 = 1095, so it appears as though the dates are encoded as an integer giving the number of days since January 1st, 1958. This is utterly bizarre, and I applaud your efforts to correct it. 

Dates should always be stored in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601). 

In this case, can you try something like:

```
babies %>%
  mutate(birthday = as.Date(date, origin = "1958-01-01"))
```

Even though this is a bizarre encoding for a date, the functions that R has for dealing with dates are always going to be better than rolling your own. 

Also, remember that since R is vectorized, you don't need to loop over the rows -- you just need to apply the right vectorized operation.